include/zim/suggestion.h

```diff
@@ -42,16 +48,29 @@ class SuggestionSearcher
     SuggestionSearcher& operator=(SuggestionSearcher&& other);
     ~SuggestionSearcher();
 
-    SuggestionSearch suggest(const Query& query);
+    SuggestionSearch suggest(const SuggestionQuery& query);
 
   private: // methods
     void initDatabase();
 
   private: // data
-    std::shared_ptr<InternalDataBase> mp_internalDb;
+    std::shared_ptr<SuggestionDataBase> mp_internalDb;
     Archive m_archive;
 };
 
+class SuggestionQuery
+{
+  public:
+    SuggestionQuery() = default;
+
+    SuggestionQuery& setVerbose(bool verbose);
```

Comment:
Why should we not keep it in the `Searcher`/`SuggestionSearcher` itself? a possible issue of keeping it in the internal db is if I make a `searcher2` from `searcher1`, due to the shared pointer `internalDb` in both the searchers, the verbosity of `searcher2` will be the same as `searcher1` even if its not set explicitly in `searcher2`. Is this something that is expected of such a feature?

Reply:
I suggest to keep it in the internal db to avoid keeping it in all the search structures (`Searcher`/`Search`/`SearcResult`/...).
But verbose is not a "real" feature of the search. It is more a debug flag. The results of the search is not dependent of this. So it is not really important where we keep it (the simpler is the better).